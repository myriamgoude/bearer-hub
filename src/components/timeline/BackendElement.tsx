import * as React from 'react'
import styled from '@emotion/styled'

import BearerLibrary from './backendElements/InstallBearerLibrary'
import BearerSDK from './backendElements/InstallBearerSDK'

const StyledDiv = styled.div<{ rightAligned: boolean }>`
  width: 50%;
  ${props => (props.rightAligned ? 'margin-left: auto;' : '')};
`

interface IBackendElementProps {
  type: string
  name: string
  rightAligned: boolean
}

const backendElementComponents: any = {
  InstallBearerLibrary: BearerLibrary,
  InstallBearerSDK: BearerSDK
}

class BackendElement extends React.Component<IBackendElementProps> {
  render() {
    const BackendElement = backendElementComponents[this.props.type]
    if (BackendElement) {
      return (
        <StyledDiv rightAligned={this.props.rightAligned}>
          <BackendElement name={this.props.name} />
        </StyledDiv>
      )
    }
    return null
  }
}

export default BackendElement
