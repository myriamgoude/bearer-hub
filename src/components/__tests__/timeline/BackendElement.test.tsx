import React from 'react'
import renderer from 'react-test-renderer'

import BackendElement from '../../timeline/BackendElement'

describe('BackendElement', () => {
  it('has no auto margin when rightAligned is false, and displays Library component for Library function type', () => {
    const tree = renderer.create(
      <BackendElement type="InstallBearerLibrary" name="connect-hello-world" rightAligned={false} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('has auto margin-left when rightAligned is true, and displays SDK content for SDK function type', () => {
    const tree = renderer.create(
      <BackendElement type="InstallBearerSDK" name="connect-hello-world" rightAligned={true} />
    )
    expect(tree).toMatchSnapshot()
  })
})
