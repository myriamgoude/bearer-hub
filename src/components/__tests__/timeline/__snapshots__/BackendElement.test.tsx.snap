// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackendElement has auto margin-left when rightAligned is true, and displays SDK content for SDK function type 1`] = `
.emotion-0 {
  font-family: 'Courier New',Courier,monospace;
}

.emotion-2 {
  width: 50%;
  margin-left: auto;
}

<div
  className="emotion-2"
>
  <h4>
    Install the Bearer SDK
  </h4>
  <div
    className="emotion-0"
  >
    $ npm install --save @bearer/react
    <br />
    // Initialize Bearer client
    <br />
    import 
    { Bearer Client }
     from '@bearer/node'
    <br />
    const bearerClient = BearerClient.new
  </div>
  <h4>
    Automate with a function call
  </h4>
  <div
    className="emotion-0"
  >
    bearerClient.call('
    ', params)
    <br />
  </div>
</div>
`;

exports[`BackendElement has no auto margin when rightAligned is false, and displays Library component for Library function type 1`] = `
.emotion-1 {
  width: 50%;
}

.emotion-0 {
  font-family: 'Courier New',Courier,monospace;
}

<div
  className="emotion-1"
>
  <h4>
    Install the Bearer Library
  </h4>
  <div
    className="emotion-0"
  >
    // from dir/
    <br />
    npm install @bearer/react
  </div>
</div>
`;
